Feature/multiple language support

Collinbrown95 opened the pull request:
Creating against issue #479 (Multiple language interface)

Using [nuxt-i18n](https://i18n.nuxtjs.org/), the nuxt.js integration for vue-i18n, for internationalization (multiple language support). This PR considers english and french, but can accommodate any number of languages.

```frontend/i18n/index.js``` defines the available locales.

Each language is given a folder based on its ISO code (e.g. ‘en’, ‘fr’). Multiple files can be contained in each folder to prevent translation files from getting too long (e.g. ```home.js``` can contain translations for the home page, ```projects.js``` could contain translations for the projects page).

Locales are accessed through the ```$t()``` method (e.g. ```$t(‘home.welcome’)```) gets the locale message for the key ```welcome``` stored in the object exported from ```home.js```.

Localized URLs are created using the ```localePath``` method.

Discussion
-----------
1. Any objections to using nuxt-i18n?
2. Are there any preferences about how to organize the files in the localization folders?

Depending on the answers to the above, I can implement the translations for English and French following the methodology outlined in this PR.

Hironsan commented:
Awesome! Please go ahead.

kuraga commented:
@Collinbrown95 , thanks very much!

Implement #605 here, please!

Collinbrown95 commented:
@kuraga Addressed #605 in commit [538159c](https://github.com/doccano/doccano/pull/976/commits/538159cfbabbc73028fecc61630247e7a9f62093).

There are a couple of things I noticed, but that seem out of scope of this PR.

First, it would be nice to have the ability to override the default locales. This [StackOverflow post](https://stackoverflow.com/a/47427963) appears to have a good suggestion, but unless there is significant demand for this feature in this PR, I will defer this to a separate issue.

Second, it appears that certain dynamic routes (both localized and non-localized) resolve to unexpected URLs. For example, ```127.0.0.1:3000/projects/1/labels``` resolves to ```127.0.0.1:3000/projects/1``` when entered directly in the address bar of the browser. I am not sure if this is intended behaviour or an error, but it leads to a specific issue when the localized ```nuxt-link``` for project pages brings the user to the default locale of that project page on their first visit (but not on subsequent visits). This is a small bug, but should be flagged nonetheless. I will start a separate issue to investigate this further.

@Hironsan apart from the above two points, this PR is ready for review. Please let me know if you would like any changes.

Hironsan commented on frontend/components/organisms/labels/LabelCreationForm.vue:
```diff
@@ -1,9 +1,9 @@
-    title="Create Label"
-    agree-text="Create"
-    cancel-text="Cancel"
+    :title="$t('projects.createLabel')"
```
Incorrect title:
![image](https://user-images.githubusercontent.com/6737785/94093352-f7218380-fe57-11ea-8e17-4a2be6324207.png)

Collinbrown95 replied:
Addressed in https://github.com/doccano/doccano/pull/976/commits/aa2679edd15b629335e74a10781123373b0e7880

Hironsan replied:
Wrong. `labels.createLabel` is correct.

Collinbrown95 replied:
My mistake. Fixed in https://github.com/doccano/doccano/pull/976/commits/5ea290cfa8b9247d4c6b6a24fb4acc22c4bc450d

Hironsan commented on frontend/components/organisms/labels/LabelCreationForm.vue:
```diff
@@ -18,19 +18,18 @@
         </v-alert>
         <v-text-field
           v-model="labelName"
           :rules="labelNameRules"
```
Maybe forget to change.

The same thing for `colorRules`:

```
...
:rules="colorRules"
...
```

Collinbrown95 replied:
Addressed in https://github.com/doccano/doccano/pull/976/commits/aa2679edd15b629335e74a10781123373b0e7880

Hironsan commented on frontend/i18n/en/rules.js:
```diff
@@ -0,0 +1,38 @@
+    colorRequired: 'Color is required'
+  },
+  labelNameRules: {
+    labelRequied: 'Label name is required',
```
Typo: `Requied` -> `Required`

Collinbrown95 replied:
Addressed in https://github.com/doccano/doccano/pull/976/commits/484ce03323a7ded5190dc6c8c425735c0a765b19

Hironsan commented on frontend/i18n/fr/rules.js:
```diff
@@ -0,0 +1,38 @@
+    colorRequired: 'La couleur est obligatoire'
+  },
+  labelNameRules: {
+    labelRequied: 'Le nom de l\'étiquette est obligatoire',
```
Typo: `Requied` -> `Required`

Collinbrown95 replied:
Addressed in https://github.com/doccano/doccano/pull/976/commits/484ce03323a7ded5190dc6c8c425735c0a765b19

Collinbrown95 commented on frontend/components/organisms/layout/LocaleMenu.vue:
```diff
@@ -0,0 +1,31 @@
+      :key="locale.code"
+    >
+      <nuxt-link
+        class="v-list-item v-list-item--link theme--dark"
```
Addressed in https://github.com/doccano/doccano/pull/976/commits/49988faca412ad8428c1b4fdfbed0800506dc643

Collinbrown95 commented:
@Hironsan I've implemented your suggested changes (see specific comments for commit links).

I have also implemented your suggestion on issue #992 to resolve the issue of dynamic routes resolving to unexpected URLs. See https://github.com/doccano/doccano/pull/976/commits/a97b6c6f542dfb4cd834c9eef727dd5046683d46 for the change (note that I also localized the ```projectRoot``` path).

Hironsan commented on frontend/middleware/check-admin.js:
```diff
@@ -1,6 +1,9 @@
-export default function({ store, route, redirect }) {
+export default async function({ app, store, route, redirect }) {
+  if (store.getters['projects/isEmpty']) {
```
Need to define the `isEmpty` getter in `store/projects.js`:

```javascript
  isEmpty(state) {
    return Object.keys(state.current).length === 0 && state.current.constructor === Object
  },
```

Collinbrown95 commented:
@Hironsan Addressed the two outstanding issues in https://github.com/doccano/doccano/pull/976/commits/1c456e2d5c69b8f4f026fa22ec73c794848668bf and https://github.com/doccano/doccano/pull/976/commits/5ea290cfa8b9247d4c6b6a24fb4acc22c4bc450d.

Hironsan commented:
Thank you for the great work.
